controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -124,46 +124,52 @@ func (r *KubeadmControlPlaneReconciler) Reconcile(req ctrl.Request) (res ctrl.Re
 	}
 
 	// Initialize the patch helper.
-	patchHelper, err := patch.NewHelper(kubeadmControlPlane, r.Client)
+	patchHelper, err := patch.NewHelper(kcp, r.Client)
 	if err != nil {
 		logger.Error(err, "Failed to configure the patch helper")
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	// Fetch the Cluster.
+	cluster, err := util.GetOwnerCluster(ctx, r.Client, kcp.ObjectMeta)
+	if err != nil {
+		logger.Error(err, "Failed to retrieve owner Cluster from the API Server")
+		return ctrl.Result{}, err
+	}
+	if cluster == nil {
+		logger.Info("Cluster Controller has not yet set OwnerRef")
+		return ctrl.Result{}, nil
+	}
+	logger = logger.WithValues("cluster", cluster.Name)
+
 	defer func() {
+		// Always attempt to update status.
+		if err := r.updateStatus(ctx, kcp, cluster); err != nil {
+			logger.Error(err, "Failed to update KubeadmControlPlane Status")
+			reterr = kerrors.NewAggregate([]error{reterr, err})
+		}
+
 		// Always attempt to Patch the KubeadmControlPlane object and status after each reconciliation.
-		if patchErr := patchHelper.Patch(ctx, kubeadmControlPlane); patchErr != nil {
-			logger.Error(patchErr, "Failed to patch KubeadmControlPlane")
-			res.Requeue = true
+		if err := patchHelper.Patch(ctx, kcp); err != nil {
+			logger.Error(err, "Failed to patch KubeadmControlPlane")
+			reterr = kerrors.NewAggregate([]error{reterr, err})
```

Comment:
You can aggregate errors like this, it'll be expanded when the caller calls `Error()`